### PR TITLE
[MRG] Better defaults

### PR DESCRIPTION
**Pull request by glouppe**
Exploration for #166, supersedes #144.

- [x] gp_minimize: default kernel?
- [x] gp_minimize: default noise level alpha for the GP? or prefer alpha=0.0 + a WhiteKernel() component?
- [x] gp_minimize: search=sampling vs search=lbfgs? fix by #172 
- [x] gp_minimize: when search=lbfgs, maxiter? n_restarts_optimizer?
- [x] gp_minimize: n_points?
- [x] forest_minimize: n_points?
- [x] forest_minimize: default base estimator?
- [x] default acquisition function? from the literature and from our tests, EI seems to be the best choice among the functions we have. 
- [x] default values for kappa and xi?

### CODE REVIEW

**Comment by glouppe** [comment]
All done, I compared with @jmetzen's BO package and with GPyOpt. Not much to change in the end. I think the main source of improvement will be to implement #173.

File: skopt/gp_opt.py [hunk 1]
```diff
@@ -78,6 +78,11 @@ def gp_minimize(func, dimensions, base_estimator=None, acq="EI", xi=0.01,
     * `base_estimator` [a Gaussian process estimator]:
         The Gaussian process estimator to use for optimization.
 
+    * `alpha` [float, default=1e-10]:
+        Value added to the diagonal of the kernel matrix during fitting.
+        Larger values correspond to increased noise level in the observations
```
Comment (betatim) [inline skopt/gp_opt.py]: "to increased" -> "to an increased"

File: skopt/gp_opt.py [hunk 2]
```diff
@@ -78,6 +78,11 @@ def gp_minimize(func, dimensions, base_estimator=None, acq="EI", xi=0.01,
...
+        and reduce potential numerical issue during fitting.
```
Comment (betatim) [inline skopt/gp_opt.py]: issueS

**Comment by betatim** [comment]
How did you pick the values? If you did any kind of benchmarking or other quantitative thing we should store those scripts/notebooks in the repo somewhere so we can use them in the future. (If this is mainly gut feeling and what others do that is also fine :))

Did you look at the examples to see if any of them are affected by the changes in defaults?

Otherwise: LGTM

**Comment by glouppe** [comment]
> How did you pick the values? If you did any kind of benchmarking or other quantitative thing we should store those scripts/notebooks in the repo somewhere so we can use them in the future. (If this is mainly gut feeling and what others do that is also fine :))

I mostly compared with other packages. I did not do a very thorough analysis to be honest. I think we have reasonably good defaults.

**Comment by glouppe** [comment]
> Did you look at the examples to see if any of them are affected by the changes in defaults?

Good point, I'll regenerate all of them.

**Comment by glouppe** [comment]
Okay hop, merging.

**Comment by betatim** [comment]
>> Did you look at the examples to see if any of them are affected by the changes in defaults?

> Good point, I'll regenerate all of them.

Worth making an issue to find out if we can have travis upload the re-rendered notebooks somewhere for each PR?

File: skopt/tree_opt.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def _tree_minimize(func, dimensions, base_estimator, n_calls,
 
 
 def gbrt_minimize(func, dimensions, base_estimator=None, n_calls=100,
-                  n_points=20, n_random_starts=10, x0=None, y0=None,
+                  n_points=1000, n_random_starts=10, x0=None, y0=None,
```
Comment (MechCoder) [inline skopt/tree_opt.py]: Since `n_points` is random anyway, why not set it to 10000 for consistency with Spearmint and SMAC?

**Comment by MechCoder** [comment]
@betatim Please do so..

**Pull request merged**